### ISSUE TITLE
[Tutorial] How to get the result data of trades.

### ISSUE THREAD

**Issue by frxncisjoseph** [n1nj4z33/iqoptionapi#17]
Hi,

I don't have the time to fork and manually make a pull request so keep this bumped.
I'll be making a couple pull requests on the weekend which add various functionalities to the API.

In order to receive the status of trades alongside the result...

1. Add the following code to on line 50 inside the 'iqoptionapi/ws/client.py' file:
```python
if message["name"] == "listInfoData":
    listinfodata = lambda: None
    listinfodata.__dict__ = message["msg"][0]
    self.api.listinfodata.add_listinfodata(listinfodata)
```

2. Create a new class in 'iqoptionapi/ws/objects' called 'listinfodata.py' and add the following code:
```python
"""Module for IQ Option Candles websocket object."""
import json

from collections import OrderedDict

from iqoptionapi.ws.objects.base import Base


class ListInfoData(Base):
    def __init__(self):
        super(ListInfoData, self).__init__()
        self.__name = "listInfoData"
        self.__listinfodata_list = OrderedDict()

    @property
    def listinfodata_list(self):
        """Property to get listinfodata list.

        :returns: The list of listinfodata.
        """
        return self.__listinfodata_list

    @listinfodata_list.setter
    def listinfodata_list(self, listinfodata_list):
        """Method to set listinfodata list."""
        self.__listinfodata_list = listinfodata_list

    @property
    def current_listinfodata(self):
        """Method to get last iteminfodata item.

         :returns: The object of listinfodata.
         """
        return self.listinfodata_list[next(reversed(self.listinfodata_list))]
    
    @property
    def get_listinfodata(self, id):
        return self.listinfodata_list[id]
    
    def add_listinfodata(self, new_listinfodata):
        """Method to add listinfodata."""
        self.listinfodata_list[new_listinfodata.id] = new_listinfodata
```

3. Add the following code as an import inside the 'iqoptionapi/api.py' file.
```python
from iqoptionapi.ws.objects.listinfodata import ListInfoData
```

4. Add the following code on line 46 inside the 'iqoptionapi/api.py' file.
```python
listinfodata = ListInfoData()
```

Once done, you can use the new function by doing the following format, i.e. You can give the result by using the 'win' variable which returns either 'equal' (if the trade is ongoing or a draw), 'win' (if the trade was a win), or 'loose' (if the trade was a loss)
```python
self.api.listinfodata.current_listinfodata.win
```
or
```python
self.api.listinfodata.get_listinfodata(foo)
```

For more information on what you can actually get, you can use this JSON data that I've extracted myself while testing.

```json
{"name":"listInfoData","msg":[{"amount":1000000,"id":2095724656,"refund":0,"currency":"USD","currency_char":"$","active_id":1,"active":"EURUSD","value":1.07736,"exp_value":1077360,"dir":"call","created":1489706346,"expired":1489706400,"type_name":"turbo","type":"front.TU","profit":100,"profit_amount":1,"win_amount":1.74,"loose_amount":0,"sum":1,"win":"equal","now":1489706346,"user_id":0,"game_state":0,"profit_income":174,"profit_return":0,"option_type_id":3,"site_id":1,"is_demo":false,"user_balance_id":0,"client_platform_id":9,"re_track":"null","params":null}]}
```

**Comment by frxncisjoseph** [comment]
If you need help then comment below and please leave a thumbs up!

**Comment by santiagonn** [comment]
Is is possible that you share your version of the api/bot with all these changes? Thanks a lot!!
Do you know if its possible to run this in a demo account?

**Comment by frxncisjoseph** [comment]
Yes, I'm current running the api against my demo account with IQ Option.
You can do this using the `self.api.change_balance('practice');` method.

I will upload my own altered version of the api publicly to my repository to be forked.

**Comment by santiagonn** [comment]
@frxncisjoseph great thanks!

**Comment by 0xAalaoui** [comment]
@frxncisjoseph can't wait to try your altered version of the api.

**Comment by crypto-maniac** [comment]
yeahhh !!!

**Comment by AllenUchiha** [comment]
great work

**Comment by crypto-maniac** [comment]
Hello Frxncisjoseph, could you please upload your altered version please?
thx you :)

**Comment by frxncisjoseph** [comment]
@crypto-maniac I have already done so, you can find it available [here](https://github.com/frxncisjoseph/iqoption-api).

I'll be updating the repository on a regular basis with newer and fulfilled features.

Feel free to make pull requests!

**Comment by crypto-maniac** [comment]
You rock !

**Comment by frxncisjoseph** [comment]
Keeping this **BUMPED** as it seems new comers are unaware of how to handle the incoming WS messages.

**Comment by Aiacos** [comment]
please make a pull request

**Comment by shiundu** [comment]
This is breaking 

>File "/Users/theHopeful/Projects/iqoptionapi/iqoptionapi/ws/objects/listinfodata.py", line 34, current_listinfodata
return self.listinfodata_list[next(reversed(self.listinfodata_list))]
StopIteration

**Comment by cz3kit** [comment]
@shiundu 
The buy order is most probably the problem. I had the same issue when my buy order did not go through.

**Comment by leonardoRampazzo** [comment]
how can i use this self.api.listinfodata.get_listinfodata(foo), to get something like this 
{"name":"listInfoData","msg":[{"amount":1000000,"id":2095724656,"refund":0,"currency":"USD","currency_char":"$","active_id":1,"active":"EURUSD","value":1.07736,"exp_value":1077360,"dir":"call","created":1489706346,"expired":1489706400,"type_name":"turbo","type":"front.TU","profit":100,"profit_amount":1,"win_amount":1.74,"loose_amount":0,"sum":1,"win":"equal","now":1489706346,"user_id":0,"game_state":0,"profit_income":174,"profit_return":0,"option_type_id":3,"site_id":1,"is_demo":false,"user_balance_id":0,"client_platform_id":9,"re_track":"null","params":null}]}

I dont know what to use as parameter self.api.listinfodata.get_listinfodata(**foo**)

**Comment by gwdm** [comment]
if you execute a buy you will get a list info message when you enter the trade and again a listinfo record at expiry of the trade
